test/PR_test/integration_test/op/tensorop/loss/test_l2_regularization.py

```diff
@@ -0,0 +1,172 @@
+        classes: The number of outputs the model should generate.
+    """
+    def __init__(self, input_shape: Tuple[int, int, int] = (1, 28, 28), classes: int = 10) -> None:
+        super().__init__()
+        self.fc1 = nn.Linear(input_shape[1] * input_shape[2], 300)
+        self.fc1.weight.data.fill_(0.01)
+        self.fc1.bias.data.fill_(0.0)
+
+        self.fc2 = nn.Linear(300, 64)
+        self.fc2.weight.data.fill_(0.01)
+        self.fc2.bias.data.fill_(0.0)
+
+        self.fc3 = nn.Linear(64, classes)
+        self.fc3.weight.data.fill_(0.01)
+        self.fc3.bias.data.fill_(0.0)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = x.view(x.size(0), -1)
+        x = fn.relu(self.fc1(x))
+        x = fn.relu(self.fc2(x))
+        x = fn.softmax(self.fc3(x), dim=-1)
+        return x
+
+
+def MyNet_tf(input_shape: Tuple[int, int, int] = (28, 28, 1), classes: int = 10) -> tf.keras.Model:
+    """A standard DNN implementation in TensorFlow.
+
+    The MyNet model has 3 dense layers.
+
+    Args:
+        input_shape: shape of the input data (height, width, channels).
+        classes: The number of outputs the model should generate.
+
+    Returns:
+        A TensorFlow MyNet model.
+    """
+    #_check_input_shape(input_shape)
+    model = Sequential()
+    model.add(layers.Flatten(input_shape=(28, 28)))
+    model.add(
+        layers.Dense(units=300,
+                     kernel_initializer=initializers.Constant(0.01),
+                     bias_initializer=initializers.Zeros(),
+                     activation='relu'))
+    model.add(
+        layers.Dense(units=64,
+                     kernel_initializer=initializers.Constant(0.01),
+                     bias_initializer=initializers.Zeros(),
+                     activation='relu'))
+    model.add(
+        layers.Dense(units=classes,
+                     kernel_initializer=initializers.Constant(0.01),
+                     bias_initializer=initializers.Zeros(),
+                     activation='softmax'))
+
+    return model
+
+
+class TestL2Regularization(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # Weight Decay pytorch model
+        cls.beta = 0.1
+
+    def test_l2_regularization_pytorch(self):
+        pytorch_l2 = fe.build(model_fn=MyNet_torch, optimizer_fn="adam")
+        l2 = L2Regularizaton(inputs='x', outputs='x', model=pytorch_l2)
+        output = l2.forward(data=torch.tensor([0.0]), state={})
+        output = output.detach()
+        self.assertTrue(np.allclose(output.numpy(), 0.12751994))
+
+    def test_l2_regularization_tensorflow(self):
+        tf_l2 = fe.build(model_fn=MyNet_tf, optimizer_fn="adam")
+        l2 = L2Regularizaton(inputs='x', outputs='x', model=tf_l2)
+        output = l2.forward(data=tf.constant([0.0]), state={})
+        self.assertTrue(np.allclose(output.numpy(), 0.1275205))
+
+    def test_pytorch_weight_decay_vs_l2(self):
```

Comment:
we'd also want to test that the tf model gets the same (within tolerance) weights